cas-server-support-osf/src/main/java/io/cos/cas/web/flow/OpenScienceFrameworkLoginHandler.java

```diff
@@ -40,17 +152,124 @@
      */
     public Event beforeLogin(final RequestContext context) {
 
-        return isInstitution(context) ? new Event(this, "institution") : new Event(this, "osf");
+        final OpenScienceFrameworkCampaign osfCampaign = getOsfCampaigns(context);
+        context.getFlowScope().put("campaign", osfCampaign.toJson());
+
+        if (osfCampaign.isInstitutionLogin()) {
```

Comment:
As discussed with Product Team, institution has high priority over campaigns. No campaign information is displayed if campaign user choose to login through institutions.